From 9ea0d4232328920add887f2df314b633b2008376 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 28 Sep 2016 16:21:17 +0300
Subject: [PATCH 0532/1345] mvpp2x: ioremap only GOP part of NetComplex

commit  dcbc79a9ab468899c6a68675d3bcd7b3a9afebc7 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Only GOP part of NetComplex ioremaped
- Remove comphy selector definitions

Change-Id: Ic89895220ccc0cdde42ca95050338dc4e083876b
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33187
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../ethernet/marvell/mvpp2x/mv_gop110_hw_type.h    |   36 +++-----------------
 1 file changed, 5 insertions(+), 31 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h
index b962b9b..542ef9e 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw_type.h
@@ -1850,13 +1850,13 @@
 
 /******************************************************************************/
 /* System Soft Reset 1 */
-#define MV_GOP_SOFT_RESET_1_REG		0x108
+#define MV_GOP_SOFT_RESET_1_REG		0x8
 
 #define NETC_GOP_SOFT_RESET_OFFSET		6
 #define NETC_GOP_SOFT_RESET_MASK	(0x1 << NETC_GOP_SOFT_RESET_OFFSET)
 
 /* Ports Control 0 */
-#define MV_NETCOMP_PORTS_CONTROL_0	(0x110)
+#define MV_NETCOMP_PORTS_CONTROL_0	(0x10)
 
 #define NETC_CLK_DIV_PHASE_OFFSET		31
 #define NETC_CLK_DIV_PHASE_MASK		(0x1 << NETC_CLK_DIV_PHASE_OFFSET)
@@ -1883,7 +1883,7 @@
 #define NETC_GOP_ENABLE_MASK		(0x1 << NETC_GOP_ENABLE_OFFSET)
 
 /* Ports Control 1 */
-#define MV_NETCOMP_PORTS_CONTROL_1	(0x114)
+#define MV_NETCOMP_PORTS_CONTROL_1	(0x14)
 
 #define NETC_PORT_GIG_RF_RESET_OFFSET(port)	(28 + port)
 #define NETC_PORT_GIG_RF_RESET_MASK(port)	\
@@ -1893,12 +1893,12 @@
 #define NETC_PORTS_ACTIVE_MASK(port)	(0x1 << NETC_PORTS_ACTIVE_OFFSET(port))
 
 /* Ports Status */
-#define MV_NETCOMP_PORTS_STATUS		(0x11C)
+#define MV_NETCOMP_PORTS_STATUS		(0x1C)
 #define NETC_PORTS_STATUS_OFFSET(port)		(0 + port)
 #define NETC_PORTS_STATUS_MASK(port)	(0x1 << NETC_PORTS_STATUS_OFFSET(port))
 
 /* Networking Complex Control 0 */
-#define MV_NETCOMP_CONTROL_0		(0x120)
+#define MV_NETCOMP_CONTROL_0		(0x20)
 
 #define NETC_GBE_PORT1_MII_MODE_OFFSET		2
 #define NETC_GBE_PORT1_MII_MODE_MASK	\
@@ -1912,32 +1912,6 @@
 #define NETC_GBE_PORT0_SGMII_MODE_MASK	\
 	(0x1 << NETC_GBE_PORT0_SGMII_MODE_OFFSET)
 
-/* ComPhy Selector */
-#define COMMON_PHYS_SELECT_REG		(0x40)
-
-#define COMMON_PHYS_SELECT_LANE_OFFSET(lane)	(4 * lane)
-#define COMMON_PHYS_SELECT_LANE_MASK(lane)    \
-	(0xF << COMMON_PHYS_SELECT_LANE_OFFSET(lane))
-#define COMMON_PHYS_SELECT_LANE_UNCONNECTED	(0x0)
-
-#define COMMON_PHYS_SELECT_LANE_0_ETH2		(0x1)
-#define COMMON_PHYS_SELECT_LANE_1_ETH3    \
-	(0x1 << COMMON_PHYS_SELECT_LANE_OFFSET(1))
-#define COMMON_PHYS_SELECT_LANE_2_ETH0    \
-	(0x1 << COMMON_PHYS_SELECT_LANE_OFFSET(2))
-#define COMMON_PHYS_SELECT_LANE_3_ETH1    \
-	(0x1 << COMMON_PHYS_SELECT_LANE_OFFSET(3))
-#define COMMON_PHYS_SELECT_LANE_3_ETH2    \
-	(0x2 << COMMON_PHYS_SELECT_LANE_OFFSET(3))
-#define COMMON_PHYS_SELECT_LANE_4_ETH2    \
-	(0x1 << COMMON_PHYS_SELECT_LANE_OFFSET(4))
-#define COMMON_PHYS_SELECT_LANE_4_ETH0    \
-	(0x2 << COMMON_PHYS_SELECT_LANE_OFFSET(4))
-#define COMMON_PHYS_SELECT_LANE_5_ETH3    \
-	(0x1 << COMMON_PHYS_SELECT_LANE_OFFSET(5))
-#define COMMON_PHYS_SELECT_LANE_5_ETH1    \
-	(0x2 << COMMON_PHYS_SELECT_LANE_OFFSET(5))
-
 /* SD1 Control1 */
 #define SD1_CONTROL_1_REG		(0x148)
 
-- 
1.7.9.5

